python-package/lightgbm/callback.py

```diff
@@ -230,10 +230,10 @@ def _callback(env):
                 continue  # train data for lgb.cv or sklearn wrapper (underlying lgb.train)
             elif env.iteration - best_iter[i] >= stopping_rounds:
                 if verbose:
-                    _log_info('Early stopping, best iteration is:\n[%d]\t%s' % (
-                        best_iter[i] + 1, '\t'.join([_format_eval_result(x) for x in best_score_list[i]])))
+                    _log_info("Early stopping, best iteration is : \n'
+                        f'[{best_iter[i] + 1}]\t{ '\t'.join([_format_eval_result(x) for x in best_score_list[i]])}")
```

Comment:
```suggestion
                    _log_info("Early stopping, best iteration is: \n"
                                    f"[{best_iter[i] + 1}]\t{'\t'.join([_format_eval_result(x) for x in best_score_list[i]])}")
```

These suggested changes should fix these linter errors from https://github.com/microsoft/LightGBM/pull/4142/checks?check_run_id=2233222418:

```text
./python-package/lightgbm/callback.py:233:65: E203 whitespace before ':'
./python-package/lightgbm/callback.py:234:25: E128 continuation line under-indented for visual indent
```

This should also help to fix some other test failures, caused by the fact that the changes in the current diff were mixing single quotes and double quotes.